packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,255 @@
+            key: string,
+            iconName: string,
+            onClick: React.EventHandler<React.MouseEvent<HTMLButtonElement>>) {
+        return (
+            <Button
+                disabled={this.props.disabled || this.props.readOnly}
+                iconName={iconName}
+                intent={this.props.intent}
+                key={key}
+                onClick={onClick}
+            />
+        );
+    }
+
+    private inputRef = (input: HTMLInputElement) => {
+        this.input = input;
+    }
+
+    private handleDecrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        const direction = -1;
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleIncrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        const direction = 1;
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (this.props.disabled || this.props.readOnly) {
+            return;
+        }
+
+        const { keyCode } = e;
+
+        let direction: number;
+
+        if (keyCode === Keys.ARROW_UP) {
+            direction = 1;
+        } else if (keyCode === Keys.ARROW_DOWN) {
+            direction = -1;
+        } else {
+            return;
+        }
+
+        // we'd like to select the field contents after running the code in this
+        // onKeyDown handler, as a UX nicety. without e.preventDefault, some
+        // hotkeys (e.g. shift + up/down, alt + up/down) will clear the selection,
+        // resulting in an inconsistent or unintuitive experience.
+        e.preventDefault();
+
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleInputChange = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        // the type definition for EventTarget (e.target) doesn't include
+        // the 'value' property out of the box, hence the janky casting
+        const nextValue = (e.target as any).value;
+        this.setState({ shouldSelectAfterUpdate : false, value: nextValue });
+
+        if (this.props.onChange) {
+            this.props.onChange(e);
+        }
+    }
+
+    private updateValue(direction: number, isAltPressed: boolean, isShiftPressed: boolean) {
+        let delta = this.props.stepSize * direction;
+
+        if (isAltPressed && isShiftPressed) {
+            return;
+        } else if (isAltPressed) {
+            delta *= this.props.minorStepSize;
+        } else if (isShiftPressed) {
+            delta *= this.props.majorStepSize;
+        }
+
+        // pretend we're incrementing from 0 if curValue isn't defined
+        const curValue = this.state.value || "0";
+
+        // truncate floating-point result to avoid precision issues when adding
+        // binary-unfriendly deltas like 0.1
+        const newValue = (!this.isValueNumeric(curValue))
+            ? 0
+            : parseFloat((parseFloat(curValue) + delta).toFixed(2));
```

Comment:
e.g. http://stackoverflow.com/questions/7312468/javascript-round-to-a-number-of-decimal-places-but-strip-extra-zeros